test/unit/math/prim/prob/neg_binomial_log_test.cpp

```diff
@@ -6,6 +6,17 @@ TEST(ProbNegBinomial, log_matches_lpmf) {
   double alpha = 1.1;
   double beta = 2.3;
 
+  long double alpha2 = 1e11;
+  long double beta2 = 1e10;
```

Comment:
Does this need to be a ```long double``` if the results are only tested against ```float``` precision? (Genuinely asking, not sure if I'm missing something here)

Reply:
The logic is that the long double reference calculations there will be super high precision, and our neg_binomial_lpmf better match it pretty well. Our neg_binomial_lpmf is doing something more complicated though (it has gamma functions and such), so it's good that it's able to keep precision.

We could turn the precision up here: https://github.com/stan-dev/math/pull/1764#issuecomment-661158937

Reply:
A very minor point: I tried using `long double` to avoid numerical issues a few times with `neg_binomial_2_lpmf` and was always disappointed - all of the issues I had were almost untouched by moving to `long double` (my _guess_ is that this was because of some exponential-like behaviour where `long double` helds only a tiny bit longer than `double`, but I don't know for sure). It certainly can help a bit, but I wouldn't put a lot of confidence in `long double` as a "safe" method.

Reply:
Mmm, we could axe the tests then since they're not doing much other than telling us the new binomial_coefficient_log is working well (that's already tested separately).

Reply:
I think it is good the tests are there, I just don't think you get a lot of benefit with `long double` over `double`, so wanted to share my experience so that you are not overconfident :-)  (but having  `long double` can't hurt)